src/main/java/org/sonar/plugins/buildbreaker/QualityGateBreaker.java

```diff
@@ -110,6 +113,16 @@ private void execute() {
     checkQualityGate(wsClient, analysisId);
   }
 
+  private String getServerUrl(Properties reportTaskProps) {
+	  String altServerUrl = settings.getString(BuildBreakerPlugin.ALTERNATIVE_SERVER_URL_KEY);
```

Comment:
Does settings.getString() not have an additional parameter override that allows specifying a default? e.g `settings.getString(BuildBreakerPlugin.ALTERNATIVE_SERVER_URL_KEY, reportTaskProps.getProperty("serverUrl")`

That would allow you to remove this method entirely.